cmd/forwarder/http.go

```diff
@@ -87,76 +97,88 @@ func extractRemoteAddr(r *http.Request) string {
 	return remoteAddr
 }
 
-func (s *httpServer) Run() error {
-	go s.awaitShutdown()
+func (s *httpServer) handleLogs(w http.ResponseWriter, r *http.Request) {
+	defer s.posts.UpdateSince(time.Now())
 
-	//FXME: check outlet depth?
-	http.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
-		defer s.healthChecks.UpdateSince(time.Now())
-		if s.isShuttingDown {
-			http.Error(w, "Shutting down", 503)
-			return
-		}
+	if s.Config.EnforceSsl && r.Header.Get("X-Forwarded-Proto") != "https" {
+		s.handleHTTPError(w, "Only SSL requests accepted", 400)
+		return
+	}
+
+	if s.isShuttingDown {
+		s.handleHTTPError(w, "Shutting down", 503)
+		return
+	}
 
-	})
+	if r.Method != "POST" {
+		s.handleHTTPError(w, "Only POST is accepted", 400)
+		return
+	}
 
-	http.HandleFunc("/logs", func(w http.ResponseWriter, r *http.Request) {
-		defer s.posts.UpdateSince(time.Now())
+	contentType := r.Header.Get("Content-Type")
+	if !s.validContentType(contentType) {
+		s.handleHTTPError(w, "Only Content-Type application/logplex-1 is accepted", 400)
```

Comment:
The error message here is misleading, given that both `application/logplex-1` and `application/msgpack` appear to be valid. Perhaps just replace this with check and subsequent error on line 149. Something like

```
if fixers[contentType] == nil {
    s.handleHTTPError(w, "Not Supported", http.StatusNotImplemented, log.Fields{"remote_addr": remoteAddr, "requestId": requestID, "logdrain_token": logplexDrainToken})
    return
}
```

and ditch the `func (s *httpServer) validContentType(ct string) bool` definition all together?

Reply:
We should definitely ditch it, I didn't want to touch the error messaging as part of this work without getting your opinion. Being able to collapse it down to just looking for a fixer is great. I'll make that change! ty, ty.